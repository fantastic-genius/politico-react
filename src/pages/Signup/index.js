import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import NavBar from '@components/NavBar';
import Button from '@components/Button';
import InputForm from '@components/InputForm';
import { signup as signupAction } from '@actions/Auth'
import { callToast } from '@components/Toast'

const Signup = ({
  fetching, error, fetched, history, ...props
}) => {
  const initialValues = {
    firstname: '',
    lastname: '',
    othername: '',
    email: '',
    phoneNumber: '',
    password: '',
    passwordConf: ''
  }

  const [values, setValues] = useState(initialValues);

  const onChange = (e) => {
    e.persist();
    setValues(prevState => ({
      ...prevState,
      [e.target.name]: e.target.value
    }));
  }

  const onSignup = (e) => {
    e.preventDefault();
    if (values.password !== values.passwordConf) {
      callToast('error', 'Password must match')
      return;
    }

    props.signupAction(values);
  }

  useEffect(() => {
    if(error){
      callToast('error', error);
    }

    if(fetched){
      callToast('success', 'Registration Successful');
      history && history.push('/vote')
    }

  }, [error, fetched])

  return (
    <div className="container home-con">
      <NavBar />
      <div className="item-box">
        <div className="header">
          <h2>Sign Up</h2>
        </div>
        <form id="signup-form" onSubmit={ onSignup }>
          <InputForm 
            id='firstname'
            type='text'
            labelName='First Name'
            placeholder='First Name'
            name='firstname'
            pattern="^[\w]{3,20}$"
            title="first name is required and must be more than 3 character"
            onChange={ onChange }
            required
            
          />
          <InputForm 
            id='othername'
            type='text'
            labelName='Other Name'
            placeholder='Other Name'
            name='othername'
            onChange={ onChange }
          />
          <InputForm 
            id='lastname'
            type='text'
            labelName='Last Name'
            placeholder='Last Name'
            name='lastname'
            pattern="^[\w]{3,20}$"
            title="first name is required and must be more than 3 character"
            onChange={ onChange }
            required
          />
          <InputForm 
            id='email'
            type='email'
            labelName='Email'
            placeholder='Email'
            name='email'
            pattern="^[\w.]+@[\w]{2,20}.[a-z]{2,10}$"
            title="must be a valid email"
            onChange={ onChange }
            required
          />
          <InputForm 
            id='phone'
            type='text'
            labelName='Phone Number'
            placeholder='08012345678'
            name='phoneNumber'
            onChange={ onChange }
            required
          />
          <InputForm 
            id='password'
            type='password'
            labelName='Password'
            name='password'
            pattern="^[\w]{8,20}$"
            title="password is required and must be more than 8 character"
            autoComplete="off"
            onChange={ onChange }
            required
          />
          <InputForm 
            id='password-conf'
            type='password'
            labelName='Confirm Password'
            name='passwordConf'
            autoComplete="off"
            onChange={ onChange }
            required
          />
          <div className="btn-con">
            <Button type='submit' id='signup' className='btn btn-success' value='Sign Up' />
          </div>
        </form>
      </div>
    </div>
  )
};

const mapStateToProps = (state) => ({
  fetching: state.auth.fetching,
  error: state.auth.error,
  fetched: state.auth.fetched,
  user: state.auth.user
});


export default connect(mapStateToProps, {signupAction})(Signup);
