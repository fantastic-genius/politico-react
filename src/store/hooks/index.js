import { useEffect } from 'react';
import  { setUser } from '@actions/Auth';

const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(setUser());
  }, []);
};

export default useSetUser;