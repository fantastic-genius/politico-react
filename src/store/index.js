import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger as logger } from 'redux-logger';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducer from './reducers';

const middleware = applyMiddleware(promise, thunk, logger());
const store = createStore(reducer, composeWithDevTools(middleware));

export default store;
