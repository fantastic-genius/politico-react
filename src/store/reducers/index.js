import { combineReducers } from 'redux';
import auth from '@reducers/Auth';
import vote from '@reducers/Vote';
import office from '@reducers/Office';

const reducers = combineReducers({
    auth,
    vote,
    office,
});

export default reducers;
