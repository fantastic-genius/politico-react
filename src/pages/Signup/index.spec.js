import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom'
import store from '@src/store';
import Signup from './index';

describe('<Signup />', () => {
  describe('Rendering', () => {
    it('should render signup page correctly', () => {
      const component = shallow(
        <Router>
          <Signup/>
        </Router>
      );
      expect(component).toMatchSnapshot();
    });

    it('should render signup page correctly woth its children', () => {
        const component = mount(
          <Router>
            <Signup store={ store }/>
          </Router>
        );
        expect(component).toMatchSnapshot();
        expect(component.find('button')).toHaveLength(3);
        expect(component.find('input')).toHaveLength(7);
    });
  });

})
