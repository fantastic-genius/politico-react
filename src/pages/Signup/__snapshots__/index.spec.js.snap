// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> Rendering should render signup page correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <BrowserRouter>
    <Connect(Signup) />
  </BrowserRouter>
</ContextProvider>
`;

exports[`<Signup /> Rendering should render signup page correctly with its children 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Signup)
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Signup
          error={null}
          fetched={false}
          fetching={false}
          signupAction={[Function]}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          user={Object {}}
        >
          <div
            className="container home-con"
          >
            <Connect(NavBar)>
              <NavBar
                isAuthenticated={false}
              >
                <header>
                  <div
                    className="app-name"
                  >
                    <Link
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <h2>
                          POLITICO
                        </h2>
                      </a>
                    </Link>
                  </div>
                  <div
                    className="sign-btn"
                  >
                    <span
                      className="home-btn"
                    >
                      <span
                        className="home-btn"
                      >
                        <Link
                          to="/signup"
                        >
                          <a
                            href="/signup"
                            onClick={[Function]}
                          >
                            <Button
                              className="btn nav-btn"
                              value="Sign Up"
                            >
                              <button
                                className="btn nav-btn"
                              >
                                Sign Up
                              </button>
                            </Button>
                          </a>
                        </Link>
                        <Link
                          to="/login"
                        >
                          <a
                            href="/login"
                            onClick={[Function]}
                          >
                            <Button
                              className="btn nav-btn"
                              value="Login"
                            >
                              <button
                                className="btn nav-btn"
                              >
                                Login
                              </button>
                            </Button>
                          </a>
                        </Link>
                      </span>
                    </span>
                  </div>
                </header>
              </NavBar>
            </Connect(NavBar)>
            <div
              className="item-box"
            >
              <div
                className="header authform-header"
              >
                <h2>
                  Sign Up
                </h2>
              </div>
              <form
                id="signup-form"
                onSubmit={[Function]}
              >
                <InputForm
                  id="firstname"
                  labelName="First Name"
                  name="firstname"
                  onChange={[Function]}
                  pattern="^[\\\\w]{3,20}$"
                  placeholder="First Name"
                  required={true}
                  title="first name is required and must be more than 3 character"
                  type="text"
                >
                  <div>
                    <label
                      htmlFor="firstname"
                    >
                      First Name
                    </label>
                    <input
                      id="firstname"
                      name="firstname"
                      onChange={[Function]}
                      pattern="^[\\\\w]{3,20}$"
                      placeholder="First Name"
                      required={true}
                      title="first name is required and must be more than 3 character"
                      type="text"
                    />
                  </div>
                </InputForm>
                <InputForm
                  id="othername"
                  labelName="Other Name"
                  name="othername"
                  onChange={[Function]}
                  placeholder="Other Name"
                  type="text"
                >
                  <div>
                    <label
                      htmlFor="othername"
                    >
                      Other Name
                    </label>
                    <input
                      id="othername"
                      name="othername"
                      onChange={[Function]}
                      placeholder="Other Name"
                      type="text"
                    />
                  </div>
                </InputForm>
                <InputForm
                  id="lastname"
                  labelName="Last Name"
                  name="lastname"
                  onChange={[Function]}
                  pattern="^[\\\\w]{3,20}$"
                  placeholder="Last Name"
                  required={true}
                  title="first name is required and must be more than 3 character"
                  type="text"
                >
                  <div>
                    <label
                      htmlFor="lastname"
                    >
                      Last Name
                    </label>
                    <input
                      id="lastname"
                      name="lastname"
                      onChange={[Function]}
                      pattern="^[\\\\w]{3,20}$"
                      placeholder="Last Name"
                      required={true}
                      title="first name is required and must be more than 3 character"
                      type="text"
                    />
                  </div>
                </InputForm>
                <InputForm
                  id="email"
                  labelName="Email"
                  name="email"
                  onChange={[Function]}
                  pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                  placeholder="Email"
                  required={true}
                  title="must be a valid email"
                  type="email"
                >
                  <div>
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                      placeholder="Email"
                      required={true}
                      title="must be a valid email"
                      type="email"
                    />
                  </div>
                </InputForm>
                <InputForm
                  id="phone"
                  labelName="Phone Number"
                  name="phoneNumber"
                  onChange={[Function]}
                  placeholder="08012345678"
                  required={true}
                  type="text"
                >
                  <div>
                    <label
                      htmlFor="phoneNumber"
                    >
                      Phone Number
                    </label>
                    <input
                      id="phone"
                      name="phoneNumber"
                      onChange={[Function]}
                      placeholder="08012345678"
                      required={true}
                      type="text"
                    />
                  </div>
                </InputForm>
                <InputForm
                  autoComplete="off"
                  id="password"
                  labelName="Password"
                  name="password"
                  onChange={[Function]}
                  pattern="^[\\\\w]{8,20}$"
                  required={true}
                  title="password is required and must be more than 8 character"
                  type="password"
                >
                  <div>
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      autoComplete="off"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      pattern="^[\\\\w]{8,20}$"
                      required={true}
                      title="password is required and must be more than 8 character"
                      type="password"
                    />
                  </div>
                </InputForm>
                <InputForm
                  autoComplete="off"
                  id="password-conf"
                  labelName="Confirm Password"
                  name="passwordConf"
                  onChange={[Function]}
                  required={true}
                  type="password"
                >
                  <div>
                    <label
                      htmlFor="passwordConf"
                    >
                      Confirm Password
                    </label>
                    <input
                      autoComplete="off"
                      id="password-conf"
                      name="passwordConf"
                      onChange={[Function]}
                      required={true}
                      type="password"
                    />
                  </div>
                </InputForm>
                <div
                  className="btn-con"
                >
                  <Button
                    className="btn btn-success"
                    id="signup"
                    type="submit"
                    value="Sign Up"
                  >
                    <button
                      className="btn btn-success"
                      id="signup"
                      type="submit"
                    >
                      Sign Up
                    </button>
                  </Button>
                </div>
              </form>
            </div>
          </div>
        </Signup>
      </Connect(Signup)>
    </Router>
  </BrowserRouter>
</Provider>
`;
